src/main/java/com/cglx/Calculator.java

```diff
@@ -31,9 +37,25 @@ public int square(int a)
         return a * a;
     }
 
-    private void readFile() {
-        FileInputStream fis = new FileInputStream(new File("TestFile.txt"));
-        
+    private List<String> readFile(String filename)
+    {
+      List<String> records = new ArrayList<String>();
+      try
+      {
+        BufferedReader reader = new BufferedReader(new FileReader(filename));
+        String line;
+        while ((line = reader.readLine()) != null)
+        {
+          records.add(line);
+        }
+        return records;
+      }
+      catch (Exception e)
+      {
+        //System.err.format("Exception occurred trying to read '%s'.", filename);
+        e.printStackTrace();
+        return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-01.ahprod.adhaven.com:9990/coding_rules#rule_key=squid%3AS1168)